gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/test/java/io/gravitee/am/gateway/handler/common/vertx/web/handler/AuthenticationFlowHandlerTest.java

```diff
@@ -1063,19 +1249,21 @@ public void shouldNotRedirectToMFAChallengePage_adaptiveMFA_3_user_stronglyAuth(
     @Test
     public void shouldRedirectToMFAChallengePage_adaptiveMFA_3_alternate_factor_id() throws Exception {
         router.route().order(-1).handler(rc -> {
+            String rule = "{#context.attributes['geoip']['country_iso_code'] == 'FR'}";
+            EnrollSettings enrollSettings = createEnrollSettings(true, false, MfaEnrollType.REQUIRED, 0, "");
+            ChallengeSettings challengeSettings = createChallengeSettings(true, MfaChallengeType.REQUIRED, rule);
```

Comment:
**todo** : based on the test case name, the test is related to adaptive MFA but the Challenge type is REQUIRED, should be CONDITIONAL